File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
using namespace std;

int solution(vector<int> nums)
{
    int N = nums.size();
    int type[200001];
    
    for(int i = 0; i < 200001; i++) {
        type[i] = 0;
    }
    
    for (int i : nums) {
        type[i] = 1;
    }
    
    int cnt = 0;
    for(int i = 1; i < 200001; i++) {
        if(type[i] == 1) {
            cnt++;
        }
    }
    
    int answer = 0;
    
    if(cnt >= N / 2) {
        answer = N / 2;
    } else {
        answer = cnt;
    }
    
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var answer = 0;
    
    const pMap = new Map();
    nums.forEach((num) => pMap.set(num, 1));
    
    answer = pMap.size < nums.length / 2 ? pMap.size : nums.length / 2;
    return answer;
}